pkg/trtl/interceptor.go

```diff
@@ -32,27 +32,36 @@ func (t *Server) interceptor(ctx context.Context, in interface{}, info *grpc.Una
 	start := time.Now()
 
 	// Check if we're in maintenance mode
-	// TODO: update to consider the Status endpoint, once it has been added
 	if t.conf.Maintenance {
+		// The only RPC we allow in maintenance mode is Status
+		if info.FullMethod == "/trtl.v1.Trtl/Status" {
+			return handler(ctx, in)
+		}
+
+		// Otherwise we stop processing here and return unavailable
 		return nil, status.Error(codes.Unavailable, "the trtl service is currently in maintenance mode")
 	}
 
-	// Fetch peer information from the TLS info.
-	var peer *PeerInfo
-	if peer, err = peerFromTLS(ctx); err != nil {
-		log.Error().Err(err).Msg("unable to retrieve remote peer info")
-		// TODO: Uncomment when mTLS is added
-		//return nil, status.Error(codes.Unauthenticated, "unable to retrieve authenticated peer information")
-	}
+	// Fetch peer information from the TLS info if we're not in insecure mode.
+	if !t.conf.MTLS.Insecure {
+		var peer *PeerInfo
+		if peer, err = peerFromTLS(ctx); err != nil {
+			return nil, status.Error(codes.Unauthenticated, "unable to retrieve authenticated peer information")
+		}
```

Comment:
Thanks for addressing this, I suppose we should create a test for `peerFromTLS` at some point.